### PR TITLE
Add annotations for Redis StatefulSet resource in values.yaml is not …

### LINKED ISSUE #40279: Add annotations for Redis StatefulSet resource in values.yaml is not possible
### Description

In values.yaml, there is no way to define annotations for **Redis StatefulSet** resource in the metadata section.

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by cybersavvydev**
Add annotations for Redis StatefulSet resource in values.yaml is not possible #40279
Closes: #40279 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: chart/templates/redis/redis-statefulset.yaml [hunk 1]
```diff
@@ -41,6 +41,9 @@ metadata:
     {{- with .Values.labels }}
       {{- toYaml . | nindent 4 }}
     {{- end }}
+  {{- if .Values.redis.annotations }}
```
Comment (jedcunningham) [inline chart/templates/redis/redis-statefulset.yaml:44]: Don't you need to add this to `values.schema.json` as well?

Can you add a test for this too?

**Review by amoghrajesh — COMMENTED** [review]
Can you please add a test somewhere in airflow/helm_tests?

File: helm_tests/airflow_aux/test_annotations.py [hunk 2]
```diff
@@ -426,3 +426,43 @@ def test_precedence(self, values, show_only, expected_annotations):
         for k, v in expected_annotations.items():
             assert k in annotations
             assert v == annotations[k]
+
+
+class TestRedisAnnotations:
+    """Tests Redis Annotations."""
+
+    @pytest.mark.parametrize(
```
Comment (jedcunningham) [inline helm_tests/airflow_aux/test_annotations.py]: No need to parametrize if you have a single case. Can you refactor to have these inline in the test?

File: chart/values.schema.json [hunk 3]
```diff
@@ -7601,6 +7601,14 @@
                     "additionalProperties": {
                         "type": "string"
                     }
+                },
+                "annotations": {
```
Comment (jedcunningham) [inline chart/values.schema.json:7615]: You should also add something to `values.yaml` :) Makes it more easily discoverable.

**Review by jedcunningham — APPROVED** [review]

**Review by amoghrajesh — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @cybersavvydev! Congrats on your first commit 🎉

**Pull request merged**